### ISSUE TITLE
Fetching only 25 Facebook Pages

### ISSUE THREAD

**Issue by GKG4** [chatwoot/chatwoot#3082]
I have more than 40 Facebook pages, however, the self hosted version is fetching only first 25 pages.
Is there any way to edit .env variable to set it to higher number?

**Comment by GKG4** [comment]
Any update on this small query, its no big deal to solve, just need to add LIMIT {variable}

**Comment by pranavrajs** [comment]
@GKG4 We didn't have a chance to test it out. We will check it out and see if we can reproduce the error. 

If you are comfortable with sending a Pull Request, please feel free to do it. We are happy to guide you.

**Comment by GKG4** [comment]
You cant reproduce as its not an error, its about default limit of fetching FB Pages. Its fetching 25 pages only, and I have 40+ pages, therefore, you need to extend FB fetch code to increase this limit.

**Comment by sojan-official** [comment]
@GKG4 its not configurable via a variable. I think we would need to paginate through all the available pages for that account than stopping after the single API call 

ref: https://github.com/chatwoot/chatwoot/blob/1761bec6152b3c7dedf0c97da4e6468914d7e881/app/controllers/api/v1/accounts/callbacks_controller.rb#L22

**Comment by GKG4** [comment]
Apart from paginate, there is an option to set limit while querying, as per the Facebook Graph Explorer. You simply need to add Limit(100) or whatever number of pages you need to fetch, this will fetch in a single call without pagination.

**Comment by GKG4** [comment]
@page_details = mark_already_existing_facebook_pages(fb_object.get_connections('me', 'accounts')) 

if we change it to 

 @page_details = mark_already_existing_facebook_pages(fb_object.get_connections('me', 'accounts.limit(100)'))


it should work, however, since I am .net developer, I cannot compile or run this code. If you can test it, much appreciated.

**Comment by GKG4** [comment]
@sojan-official Is there any chance that you will be fixing this issue?
Because this is the only thing stopping me from using ChatWoot at the moment

I have few suggestions and if you are willing to discuss a bit, I am more than happy to contribute Open Source Windows version for Chatwoot, and probably Mac Version too.

**Comment by sojan-official** [comment]
@GKG4 Yes, We will be prioritizing it for our next week's issue intake. 

We are always open to feedback. Please feel free to initiate a GitHub discussion/issue anytime for anything :) 
We are also available on discord: https://discord.gg/cJXdrwS

**Comment by tejaswinichile** [comment]
@GKG4 We don't restrict the number of Facebook pages in chatwoot. It's not related to the pagination or limit.

fb_object.get_connections('me', 'accounts') = returns all the pages connected your account, this is the command that we run and send it to facebook to get the connected pages.

The issue might be with the access permission that your account has to the other pages. https://developers.facebook.com/docs/facebook-login/permissions/overview/#granularpermissions 

If you have configured this on the self-hosted version, and you have access to https://developers.facebook.com/tools/explorer/  then you can run the query `me/accounts` and check the connected pages to the account like below.
<img width="1367" alt="Screenshot 2021-10-04 at 6 28 15 PM" src="https://user-images.githubusercontent.com/10827585/135855553-8c44839b-a6a7-4a26-95e5-c71af37dcb57.png">


Or you can go to https://business.facebook.com/latest/settings/business_assets to check the pages to which you have access.

cc @sojan-official

**Comment by GKG4** [comment]
@sojan-official Its not you who is putting the restrictions.
Its that if you do not set fetch limit, default number of pages FB send is 25. You need to provide either pagination or option to call more than 25 pages by adding limit variable into the queries.

I already have a custom app built in .net fetching all 40+ pages, therefore, I know that I have access to those pages.

**Comment by sojan-official** [comment]
@GKG4 Thanks for posting back. Nice Hack. 
We will prioritize a proper fix :)

**Comment by GKG4** [comment]
@sojan-official 
As of now, we have 24 hours window to reply, after that we cant reply to FB Messages
However, Facebook to give option for 7 days window, if we use `HUMAN_AGENT` tag

Is it possible, if you guide me, where I can add all these, I am willing to contribute?
Also, I am willing to add different message templates as per Facebook API

**Comment by GKG4** [comment]
@sojan-official Since HUMAN_AGENT do not require any additional permission, and since Chatwoot is usually human operated only, we can make it defafult as well.

**Comment by mredodos** [comment]
some news?

**Comment by kamonwatroj** [comment]
Hi,
In chatwoot page, I still get only 25 pages, even if I select all the pages on facebook integration page.
Do we have a fix for this?

**Comment by kamonwatroj** [comment]
> [@GKG4](https://github.com/GKG4) Thanks for posting back. Nice Hack. We will prioritize a proper fix :)

Hi, @sojan-official 
In chatwoot page, I still get only 25 pages, even if I select all the pages on facebook integration page.
Do we have a fix for this?
